extensions/script/python/ExecutePythonProcessor.cpp

```diff
@@ -50,39 +50,72 @@ core::Property ExecutePythonProcessor::ModuleDirectory(core::PropertyBuilder::cr
   ->withDefaultValue("")
   ->build());
 
-core::Relationship ExecutePythonProcessor::Success("success", "Script successes");
-core::Relationship ExecutePythonProcessor::Failure("failure", "Script failures");
+core::Relationship ExecutePythonProcessor::Success("success", "Script succeeds");
+core::Relationship ExecutePythonProcessor::Failure("failure", "Script fails");
 
 void ExecutePythonProcessor::initialize() {
-  setSupportedProperties({
-    ScriptFile,
-    ScriptBody,
-    ModuleDirectory
-  });
-  setAcceptAllProperties();
-  setSupportedRelationships({
-    Success,
-    Failure
-  });
-}
+  if (getProperties().empty()) {
+    setSupportedProperties({
+      ScriptFile,
+      ScriptBody,
+      ModuleDirectory
+    });
+    setAcceptAllProperties();
+    setSupportedRelationships({
+      Success,
+      Failure
+    });
+  }
+
+  if (processor_initialized_) {
+    logger_->log_debug("Processor has already been initialized, returning...");
+    return;
+  }
 
-void ExecutePythonProcessor::onSchedule(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSessionFactory>& /*sessionFactory*/) {
   python_logger_ = core::logging::LoggerFactory<ExecutePythonProcessor>::getAliasedLogger(getName());
 
   getProperty(ModuleDirectory.getName(), module_directory_);
-
   appendPathForImportModules();
-  loadScript();
 
-  if (script_to_exec_.empty()) {
-    throw std::runtime_error("Neither Script Body nor Script File is available to execute");
+  try {
+    loadScript();
+  } catch(const std::runtime_error&) {
+    logger_->log_warn("Could not load python script while initializing. In case of non-native python processor this is normal and will be done in the schedule phase.");
```

Comment:
what makes a non-native processor fail here but succeed in the schedule?

Reply:
Normally in the `ExecutePythonProcessor`'s initialize phase the values of the properties are not yet available like in any other processor. However when we use a native python processor, after the properties are initialized the `PythonObjectFactory` sets the `ScriptFile` property's value to the python processor's file path and calls the `initialize` again and the script will be loaded.

Reply:
and a native processor's `onSchedule` won't be called, an initialization in `onSchedule` would be too late for them?

Reply:
`onSchedule` runs for that as well, but as we need to call the native processor's `describe` before the `onSchedule` so we can provide manifest our processor identity on C2 as well.

Reply:
I see, that makes sense, could you add this reason as a comment?

Reply:
Sure, added comment in e32ad12483fe66bc7759237495e9922737a3958b